lib/benchee/conversion/memory.ex

```diff
@@ -46,6 +46,52 @@ defmodule Benchee.Conversion.Memory do
               }
   }
 
+  # Unit conversion functions
+
+  @type unit_atom :: :byte | :kilobyte | :megabyte | :gigabyte | :terabyte
+  @type value_with_unit :: {number, Unit.t}
+  @type value_with_unit_atom :: {number, unit_atom}
+
+  @doc """
+  Converts a value for a specified unit atom and converts it to the equivalent of another unit of measure.
+
+  ## Examples
+
+    iex> {value, unit_atom} = Benchee.Conversion.Memory.convert({1024, :kilobyte}, :megabyte)
+    iex> value
+    1.0
+    iex> unit_atom
+    :megabyte
+  """
+  @spec convert(value_with_unit_atom, unit_atom) :: value_with_unit_atom
+  def convert({value, current_unit_atom}, desired_unit_atom) when is_atom(current_unit_atom) and is_atom(desired_unit_atom) do
+    current_value_unit = {value, unit_for(current_unit_atom)}
+    desired_unit = unit_for(desired_unit_atom)
+    {new_value, _} = convert_with_unit(current_value_unit, desired_unit)
+    {new_value, desired_unit_atom}
```

Comment:
With this approach still this could also just end in a call to `convert_with_unit(current_value_unit, desired_unit)` as it does already what you want :)

Reply:
Ah I see now what you did... imo we should always return a unit struct, the unit atoms are just meant for a nicer people interface so that people can use the atoms if they want to, but our methods whould always return the proper unit structs